src/pgduckdb_options.cpp

```diff
@@ -145,6 +190,55 @@ DuckdbInstallExtension(Datum name) {
 	return true;
 }
 
+static bool
+CanCacheRemoteObject(std::string remote_object) {
```

Comment:
I'd be more comfortable with this:
```c++
static bool
CanCacheRemoteObject(std::string remote_object) {
    return (remote_object.rfind("https://", 0) == 0) ||
           (remote_object.rfind("http://", 0) == 0) ||
           (remote_object.rfind("s3://", 0) == 0) ||
           (remote_object.rfind("s3a://", 0) == 0) ||
           (remote_object.rfind("s3n://", 0) == 0) ||
           (remote_object.rfind("gcs://", 0) == 0) ||
           (remote_object.rfind("gs://", 0) == 0) ||
           (remote_object.rfind("r2://", 0) == 0);
}
```

Multiplying `std::string::npos` multiple times feels problematic